app/models/maestrano/connector/rails/concerns/entity.rb

```diff
@@ -154,13 +158,15 @@ def public_external_entity_name
   #                 Mapper methods
   # ----------------------------------------------
   # Map a Connec! entity to the external model
-  def map_to_external(entity)
+  def map_to_external(entity, first_time_mapped = nil)
+    return self.class.creation_mapper_class.normalize(entity).with_indifferent_access if first_time_mapped
     self.class.mapper_class.normalize(entity).with_indifferent_access
   end
 
   # Map an external entity to Connec! model
-  def map_to_connec(entity)
-    mapped_entity = self.class.mapper_class.denormalize(entity).merge(id: self.class.id_from_external_entity_hash(entity))
+  def map_to_connec(entity, first_time_mapped = nil)
+    mapped_entity = self.class.creation_mapper_class.denormalize(entity).merge(id: self.class.id_from_external_entity_hash(entity)) if first_time_mapped
```

Comment:
can you refactor the `.merge(id: self.class.id_from_external_entity_hash(entity))` part?